internal/outagelog/outagelog.go

```diff
@@ -0,0 +1,46 @@
+package outagelog
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+)
+
+type Handler struct {
+	logger     *slog.Logger
+	fileHandle *os.File
+	filePath   string
+}
+
+func NewHandler(filePath string, logger *slog.Logger) (*Handler, error) {
+	fileHandle, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
```

Comment:
why not injecting this? rather than having a constructor with side effects like opening a file?

Reply:
if this is used only for testing, it may not be needed though

Reply:
I preferred to encapsulate the mechanisms for opening and closing the logfile within the package, rather than include this quite specific open call in main and in every test

Reply:
Yes, if this is used only for testing, that makes sense. But if is used in production code, thats's not necessarily a good thing.
The client is already passing the path, so you are still delegating to the client to choose where. Constructors free of side effects (IO) is often a good rule of thumb

Reply:
Would it make sense if I added an `Open` function to the package which just does the IO part, so I would first get the filehandle from there and then pass it to the constructor? It seems better to have a single function that performs the `OpenFile` call and specifies the permissions

Reply:
yes, that also works I think

Reply:
ok I've made that change